flink-core/src/main/java/org/apache/flink/core/io/PostVersionedIOReadableWritable.java

```diff
@@ -62,16 +62,23 @@ public void write(DataOutputView out) throws IOException {
 	 */
 	public final void read(InputStream inputStream) throws IOException {
 		byte[] tmp = new byte[VERSIONED_IDENTIFIER.length];
-		inputStream.read(tmp);
+		int totalRead = 0;
```

Comment:
nit: just for extra readability, could you refactor this reading logic to a separate method?

Reply:
@igalshilman just reminded me:
even better, can you add this read logic as a new `tryReadFully` method to `IOUtils`? That would be a good place for this to be.

Reply:
Moved to org.apache.flink.util.IOUtils, added a few UT